### PR TITLE
Gitian OS X

### PR DESCRIPTION
**Pull request by josephbisch**
@droark 

These are some changes to depends to try to get the OS X Gitian descriptor working. I realized I never committed the OS X descriptor, so this branch was originally intented to commit the OS X descriptor. Currently it does not include the OS X descriptor, because I haven't been able to fully test it due to the depends system using Python 3, but the Armory code still using Python 2. That causes a conflict when building _CppBlockUtils.so. However, the Gitian build does get as far as building _CppBlockUtils.so, so I can just commit the descriptor now if you want.

I also included a commit from theuni that I got from the Bitcoin Core repo. I was doing it to try to resolve an issue I was having (turned out to not fix the issue, something else did), but I'll leave it, because I figure it is good to try to keep up with relevant changes to the depends system in Bitcoin Core.

### CODE REVIEW

File: depends/packages/native_qt.mk [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Pretty much taken verbatim from Bitcoin Core
```
Comment (droark) [inline depends/packages/native_qt.mk]: I can't remember, was there a reason to stick with 5.2.1, other than it's known to work with Gitian?

File: depends/patches/native_qt/qt5-yosemite.patch [hunk 2]
```diff
@@ -0,0 +1,54 @@
+Needs to be reviewed once OS X support is added.
```
Comment (droark) [inline depends/patches/native_qt/qt5-yosemite.patch]: If we upgrade past 5.2.1 at some point, this patch can, in all likelihood, be removed. The code's in 5.4.1, for example.

File: depends/patches/native_qt/qt5-tablet-osx.patch [hunk 3]
```diff
@@ -0,0 +1,22 @@
+Needs to be reviewed once OS X support is added.
```
Comment (droark) [inline depends/patches/native_qt/qt5-tablet-osx.patch]: If we upgrade past 5.2.1 at some point, this patch can, in all likelihood, be removed. The code's in 5.4.1, for example.

File: depends/packages/native_pyqt5.mk [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package=native_pyqt5
+$(package)_version=5.4.1
+$(package)_download_path=http://sourceforge.net/projects/pyqt/files/PyQt5/PyQt-$($(package)_version)
+$(package)_file_name=PyQt-gpl-$($(package)_version).tar.gz
+$(package)_sha256_hash=c6c33f392c0a2b0c9ec16531dc88fa9aee853e9fff10f0ad9d0e4777629fe79e
+$(package)_dependencies=native_python3 native_qt native_sip
+
+define $(package)_config_cmds
+  $($(package)_prefix)/bin/python3 configure.py --confirm-license --bindir=$($(package)_prefix)/bin --qmake=$($(package)_prefix)/bin/qmake-native --sip=$($(package)_prefix)/bin/sip --sip-incdir=$($(package)_prefix)/include/python3.4m --spec=linux-g++ --verbose
```
Comment (droark) [inline depends/packages/native_pyqt5.mk]: Is the spec (linux-g++) correct?

**Comment by droark** [comment]
Thanks. Overall, I think this looks fine, although I do have a few questions. I've posted most of them. I'm also curious what it is that you added from theuni's work.

File: depends/packages/native_qt.mk (hunk 1 above)
Comment (josephbisch) [inline depends/packages/native_qt.mk]: The reason is pretty much that it was already known to work with Gitian due to Bitcoin Core using it. There is [someone working on moving Bitcoin Core to Qt 5.4.2](https://github.com/bitcoin/bitcoin/issues/6267) and that person should open a PR for Bitcoin Core any day now.

File: depends/packages/native_pyqt5.mk (hunk 4 above)
Comment (josephbisch) [inline depends/packages/native_pyqt5.mk]: Probably not entirely. I think the reason it works is that I am on Linux and this is a native package. Also Gitian uses Linux, so there are no problems with this in that case. But if you are using the depends system for a normal build and you are on OS X it probably won't work. So I'll fix that.

File: depends/packages/native_qt.mk (hunk 1 above)
Comment (josephbisch) [inline depends/packages/native_qt.mk]: We should follow [this PR to upgrade Bitcoin Core depends to Qt 5.4.2](https://github.com/bitcoin/bitcoin/pull/6314) and pull it into Armory once it is finalized.

Comment (droark) [inline depends/packages/native_qt.mk]: Thanks. I've subscribed and will keep an eye on the PR.

**Comment by droark** [comment]
Closing out per our discussion. The work will be archived for future usage. Thanks!

**Pull request closed without merging**